NeutronStandard/Sniffs/Namespaces/DisallowUnusedImportSniff.php

```diff
@@ -0,0 +1,29 @@
+<?php
+
+namespace NeutronStandard\Sniffs\Namespaces;
+
+use NeutronStandard\SniffHelpers;
+use PHP_CodeSniffer\Sniffs\Sniff;
+use PHP_CodeSniffer\Files\File;
+
+class DisallowUnusedImportSniff implements Sniff {
+	public function register() {
+		return [T_USE];
+	}
+
+	public function process(File $phpcsFile, $stackPtr) {
+		$tokens = $phpcsFile->getTokens();
+		$semiPtr = $phpcsFile->findNext([T_SEMICOLON], $stackPtr + 1);
+		if (! $semiPtr) {
```

Comment:
I don't understand how this detects if we should continue with the sniff or not. Can you expand a bit on the logic?

Reply:
If for some weird reason there's no semicolon in the file (like if we're linting an incomplete file with just the word `use`) then there's no reason to continue. 

More importantly, if we do continue, weird things could happen because `$semiPtr` will be `false`, and PHP is totally fine with doing `(false - 1)`, meaning the next line will be evaluated as `$tokens[-1]`. In this case that would just throw an error, but in some places I add to the index instead. If it was `(false + 1)`, then the next expression would be `$tokens[1]` which _does_ exist. That causes bugs that are hard to track down.

Reply:
I understand now. I have missed the part the the sniff is registered only for `T_USE`, and it did not make sense without that :)